platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/address/AddressBookInitializer.java

```diff
@@ -133,8 +118,8 @@ public AddressBookInitializer(
                 platformContext.getConfiguration().getConfigData(AddressBookConfig.class);
         this.initialState = Objects.requireNonNull(initialState, "The initialState must not be null.");
 
-        this.stateAddressBook = RosterUtils.buildAddressBook(RosterRetriever.retrieveActiveOrGenesisRoster(
-                (State) initialState.getState().getSwirldState()));
+        final var book = stateBookFn.apply(initialState);
+        this.stateAddressBook = (book == null || book.getSize() == 0) ? null : book;
```

Comment:
I understand the addition of the ternary `?:` logic. However, I don't understand why the `stateBookFn` was introduced. I've re-read the code changes in all the relevant places, including the ServicesMain, and I don't see how it's different from what was happening before when we called the RosterRetriever directly and explicitly here using the passed in `initialState`. This modification only adds an extra, seemingly completely unnecessary argument to the constructor and complicates the code, w/o adding any extra value, or at least I was unable to see it.

Could we consider nuking the `stateBookFn` and only adding the extra ternary logic when assigning a value to the instance variable here? This would simplify the code per se, as well as help reduce the size of the PR a tiny bit.

Reply:
Ah yes I added this before last PR, it's now unnecessary; will remove now.

Reply:
Removed.